src/pilot/templates/first-boot-ovs-dpdk.yaml

```diff
@@ -0,0 +1,450 @@
+# Copyright (c) 2016 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+heat_template_version: 2014-10-16
+
+parameters:
+  ComputeKernelArgs:
+    description: >
+      Space seprated list of Kernel args to be update to grub.
+      The given args will be appended to existing args of GRUB_CMDLINE_LINUX in file /etc/default/grub
+      Example: "intel_iommu=on default_hugepagesz=1GB hugepagesz=1G hugepages=1"
+    type: string
+    default: ""
+  NeutronDpdkSocketMemory:
+    description: Memory allocated for each socket
+    default: ""
+    type: string
+  NeutronVhostuserSocketDir:
+    description: The vhost-user socket directory for OVS.
+    default: ""
+    type: string
+  HostCpusNumber:
+    description: >
+      Number of logical cores to be used by ovs-dpdk processess 
+      (dpdk-lcore-mask). Comma separated list of number of cores to be used for 
+      host os on each NUMA Node. 
+    type: string
+    default: "4,4"
+    constraints:
+      - allowed_pattern: "[0-9,]+"
+  DpdkInterfaces:
+    description: Interfaces that are used by OVS-DPDK
+    type: string
+  VmWeight:  
+    description: >
+      Used along with PmdWeight, to divide the cores into two groups. For every 
+      'N' number of cores assigned for VM, 'M' cores are assigned to PMD list. 
+    type: number
+    default: 4
+  PmdWeight:  
+    description: >
+      Used along with VmWeight, to divide the cores into two groups. For every 
+      'N' number of cores assigned for VM, 'M' cores are assigned to PMD list. 
+    type: number
+    default: 1
+    
+resources:
+  userdata:
+    type: OS::Heat::MultipartMime
+    properties:
+      parts:
+      - config: {get_resource: wipe_disk}
+      - config: {get_resource: generate_parameters}
+      - config: {get_resource: set_ovs_config}
+      - config: {get_resource: set_dpdk_params}
+      - config: {get_resource: install_tuned}
+      - config: {get_resource: compute_kernel_args}
+
+  wipe_disk:
```

Comment:
Is there some way to pull in the existing wipe_disk code instead of duplicating it here?

Reply:
I'm not aware of any technical method to call this wipe_disk code outside of its underlying template. I don't think we can use it as a regular python method if that's what you meant.